kong/globalpatches.lua

```diff
@@ -288,37 +288,37 @@ return function(options)
     local old_tcp_connect
     local old_udp_setpeername
 
-    local function tcp_resolve_connect(sock, host, port, sock_opts)
+    local function strip_nils(first, second)
+      if second then
+        return first, second
+      elseif first then
+        return first
+      end
+    end
+  
+    -- need to do the extra check here: https://github.com/openresty/lua-nginx-module/issues/860
+    local function resolve_connect(f, sock, host, port_or_opts, opts)
+      if sub(host, 1, 5) == "unix:" then
+        opts = port_or_opts
+        return f(sock, host, strip_nils(opts))
+      end
+    
+      local port = port_or_opts
       local target_ip, target_port = toip(host, port)
       if not target_ip then
-        return nil, "[toip() name lookup failed]: " .. tostring(target_port) -- err
+        return nil, "[toip() name lookup failed]: " .. tostring(target_port)
       end
-
-      -- need to do the extra check here: https://github.com/openresty/lua-nginx-module/issues/860
-      if not sock_opts then
-        return old_tcp_connect(sock, target_ip, target_port)
-      end
-
-      return old_tcp_connect(sock, target_ip, target_port, sock_opts)
+      return f(sock, target_ip, strip_nils(target_port, opts))
```

Comment:
the code is correct, but I think you're overdoing the variable naming to explain some intricate details of how the socket functions work, which complicates the code and reduces code readability. Can't we just drop the local variables and stick to:
```lua
    local function resolve_connect(f, sock, host, port, opts)
      if sub(host, 1, 5) ~= "unix:" then
        host, port = toip(host, port)
        if not host then
          return nil, "[toip() name lookup failed]: " .. tostring(port)
        end
      end

      return f(sock, host, strip_nils(port, opts))
    end
```

Reply:
seems ok for me too. in the end, this is a patch and users might not expect that the original ngx.socket.* are modified by kong, thus they will probably lookup the original ngx lua to find the definition.